packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -421,9 +432,46 @@ class ReportData {
     });
     this.reset();
   }
+
   reset() {
     this.report = new Report({ header: this.header });
     this.size = 0;
+    this.traceCache = new Map<TraceCacheKey, boolean>();
+  }
+}
+
+class StatsMap {
+  readonly map: Map<IStatsContext, ContextualizedStats> = new Map<
```

Comment:
similarly this needs to JSON-serialize (or whatever) the key, which I suspect means you could use some more tests...